src/core/timer.c

```diff
@@ -380,6 +380,9 @@ static void timer_enter_waiting(Timer *t, bool time_change) {
                         if (r < 0)
                                 continue;
 
+                        if (t->persistent && v->next_elapse < ts.realtime)
+                                v->next_elapse = ts.realtime;
```

Comment:
i think a comment here would be good. Also, I think we should probably slightly alter this: instead of comparing the current time, compare it with the time systemd was first started, i.e. `UNIT(t)->manager->timestamps[MANAGER_TIMESTAMP_USERSPACE].realtime`. This is very similar to using the current time, but means that reloading pid 1 during runtime won't cause this clamping to happen anymore.

Reply:
by using `UNIT(t)->manager->timestamps[MANAGER_TIMESTAMP_USERSPACE].realtime` we also don't have to check `t->persistent` anymore, since the logic is true implicitly for non-persistent timers of course.

Reply:
Well, if we use ```UNIT(t)->manager->timestamps[MANAGER_TIMESTAMP_USERSPACE].realtime```, it seems that it doesn't work for suspend/hibernate.
Is it okay?

Reply:
hmm, i'd prefer to ignore the suspend/hibernate case for this for now. If people show up who actually have issues with that we can revisit this after all

Reply:
I see, updated!

Reply:
> hmm, i'd prefer to ignore the suspend/hibernate case for this for now. If people show up who actually have issues with that we can revisit this after all

As mentioned in issue #21166, I certainly had trouble with the timers expiring immediately after resume. Consider I have like 5-10 lxd containers running, and every time I resumed, they and the host OS started downloading updates and installing them all at the same time. It was pure misery, so I now run the laptop 24 hours a day.